Automattic/wp-calypso
# Badge component: Temporarily wrap old Badge component with the new one

Related to #77564 

## Proposed Changes

* At @daledupreez's suggestion, point the old `Badge` component to the new one, allowing us to transition from the old code to the new in smaller pieces and without risking updates to the old component in the interim.
* Remove old docs and files.
* Update instances of the Badge examples to point to the new component
* Add correct typing to the `type` property on some instances to fix type errors 

Some visual examples of this component in action:

<img width="383" alt="Screenshot 2023-07-12 at 10 00 54 AM" src="https://github.com/Automattic/wp-calypso/assets/2124984/9a678b4a-23de-4039-a75a-325f27bfc051">
<img width="678" alt="Screenshot 2023-07-12 at 10 01 47 AM" src="https://github.com/Automattic/wp-calypso/assets/2124984/623cb17a-52fa-4784-a83d-9c2aae813d56">
<img width="1064" alt="Screenshot 2023-07-12 at 11 05 09 AM" src="https://github.com/Automattic/wp-calypso/assets/2124984/e6c3ff3d-6910-4e8c-b9f0-3b5acbf641d7">


## Testing Instructions



* Switch to this PR
* Test instances of the Badge component; examples can be found in `devdocs/design/badge`, check the Domains screens at `/manage/domains`, check site setup at `/start`

## Pre-merge Checklist
